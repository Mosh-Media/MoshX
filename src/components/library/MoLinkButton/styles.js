const styles = theme => ({
  link: {
    backgroundColor: theme.palette.primary.main,
    textDecoration: "none",
    padding: "12px 22px",
    color: theme.white?.medium,
    fontSize: "1.1rem",
    borderRadius: theme.space.xs,
    textTransform: "initial",
    fontWeight: "600",
    "&:hover": {
      background: theme.grey?.dark
    },
    "&:hover .arrow": {
      transform: "translateX(5px)"
    }
  },
  icon: {
    transition: "transform .35s",
    paddingLeft: "4px",
    paddingTop: "2px"
  }
});

export default styles;
