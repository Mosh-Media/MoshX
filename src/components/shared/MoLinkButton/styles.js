const styles = (theme) => ({
  link: {
    backgroundColor: theme.blue?.medium,
    textDecoration: "none",
    padding: "12px 22px",
    color: theme.white?.medium,
    fontSize: "1.1rem",
    borderRadius: "3px",
    textTransform: "initial",
    fontWeight: "600",
    "&:hover": {
      background: theme.grey?.dark,
    },
    "&:hover arrow": {
      transform: "translateX(5px)",
    },
  },
  icon: {
    paddingLeft: "4px",
  },
});

export default styles;
