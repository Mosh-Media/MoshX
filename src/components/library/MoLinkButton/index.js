import React from "react";

import ArrowForwardIcon from "@material-ui/icons/ArrowForward";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";
import withStyles from "@material-ui/core/styles/withStyles";
import styles from "./styles";
import PropTypes from "prop-types";

const MoLinkButton = ({
  classes,
  children,
  handleButtonClick,
  href,
  isArrowIcon,
  text,
  size,
  variant
}) => {
  /* Make sure href isn't set to prevent router link errors */
  const hrefProp = href || "#";

  /* Prevent errors when handleButtonClick not available  */
  const handleOnCLick = e => {
    return handleButtonClick ? handleButtonClick(e) : null;
  };

  return (
    <Button
      onClick={e => handleOnCLick(e)}
      component={Link}
      to={hrefProp}
      className={classes.button}
      color={"primary"}
      size={size}
      variant={variant}
      endIcon={
        isArrowIcon && <ArrowForwardIcon className={`${classes.icon} arrow`} />
      }
    >
      {text} {children}
    </Button>
  );
};

MoLinkButton.propTypes = {
  classes: PropTypes.object,
  children: PropTypes.any,
  handleButtonClick: PropTypes.func,
  href: PropTypes.string,
  isArrowIcon: PropTypes.bool,
  text: PropTypes.string,
  color: PropTypes.oneOf(["primary", "secondary", "default", null]),
  size: PropTypes.oneOf(["small", "medium", "large", null]),
  variant: PropTypes.oneOf(["contained", "outlined", "text", null])
};

export default withStyles(styles)(MoLinkButton);
