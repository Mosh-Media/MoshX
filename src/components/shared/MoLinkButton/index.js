import React from "react";

import ArrowForwardIcon from "@material-ui/icons/ArrowForward";
import Button from "@material-ui/core/button";
import { Link } from "react-router-dom";
import styles from "./styles";
import withStyles from "@material-ui/core/styles/withStyles";

const MoLinkButton = ({ classes, href, text }) => {
  return (
    <Button component={Link} to={href ? href : "#"} className={classes.link}>
      {text}
      <ArrowForwardIcon
        className="arrow"
        style={{ fontSize: "1em", paddingLeft: "4px" }}
      />
    </Button>
  );
};

export default withStyles(styles)(MoLinkButton);
