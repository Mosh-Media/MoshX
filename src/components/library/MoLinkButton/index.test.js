import React from "react";

import ShallowRenderer from "react-test-renderer/shallow";
import MoLinkButton from "./index.js";

const renderer = new ShallowRenderer();

describe("MoLinkButton Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<MoLinkButton />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
