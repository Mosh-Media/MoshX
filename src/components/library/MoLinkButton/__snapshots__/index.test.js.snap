// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoLinkButton Component should match the snapshot 1`] = `
<MoLinkButton
  classes={
    Object {
      "icon": "MoLinkButton-icon-2",
      "link": "MoLinkButton-link-1",
    }
  }
/>
`;
